src/components/gamespec/Board.js

```diff
@@ -11,15 +11,50 @@ const boxTarget = {
       let offset = monitor.getClientOffset(),
           item = monitor.getItem();
 
+      let pieces = Object.assign([], props.getPieces());
       let items = props.getItems();
+      let copy = Object.assign([], items);
       let rect = component.refs.stage.getStage().getContainer().getBoundingClientRect();
       offset.x = offset.x - rect.left;
       offset.y = offset.y - rect.top;
       let image = item.image;
-      items.push({image, offset});
-      props.setItems(items);
+      let indice = [];
+      let pieceIdx = [];
+      let flag = true;
+      if(copy.length !== 0){
+        for(let j = 0; j < copy.length && flag ; j++){
+          let off = copy[j].offset;
+          let x = off.x - offset.x
+          let y = off.y - offset.y
+          if(x * x < Math.pow(copy[j].image.width / 5, 2) && y * y < Math.pow(copy[j].image.height / 5, 2)){
+            let index = copy[j].indice[0];
+            indice.push(index);
+            indice.push(pieces[index].pieceIdx - 1);
+            copy.push({image, offset, indice})
+            pieces[index].pieceIdx.push(copy.length-1);
```

Comment:
Can you use better variable names, the code is very difficult to read at the moment to understand what is going on

Reply:
Sure, I will try to rename them. Maybe some annotation is useful here?